x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformConfig.java

```diff
@@ -381,4 +380,114 @@ public static DataFrameTransformConfig fromXContent(final XContentParser parser,
 
         return lenient ? LENIENT_PARSER.apply(parser, optionalTransformId) : STRICT_PARSER.apply(parser, optionalTransformId);
     }
+
+    public static class Builder {
+        private String id;
+        private SourceConfig source;
+        private DestConfig dest;
+        private TimeValue frequency;
+        private SyncConfig syncConfig;
+        private String description;
+        private Map<String, String> headers;
+        private Version transformVersion;
+        private Instant createTime;
+        private PivotConfig pivotConfig;
+
+        public Builder() { }
+
+        public Builder(DataFrameTransformConfig config) {
+            this.id = config.id;
+            this.source = config.source;
+            this.dest = config.dest;
+            this.frequency = config.frequency;
+            this.syncConfig = config.syncConfig;
+            this.description = config.description;
+            this.headers = config.headers;
```

Comment:
I think we should not copy headers, but always let them set as the config potentially comes from another user and if you forget to overwrite you can create a config as another user.

Reply:
Yes, definitely we must take the headers from the update request, not the previous config.  This will then match the behaviour for datafeeds - see the `IMPORTANT` note in https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-update-datafeed.html

Also see the bit of https://www.elastic.co/guide/en/elastic-stack-overview/current/ml-dfeeds.html that says:

> One way to update the roles that are stored within the datafeed without changing any other settings is to submit an empty JSON document (`{}`) to the update datafeed API.

For consistency the same should be possible with this API.

Reply:
Currently, this API sets the headers to the user who calls `_update` for parity with the datafeeds API. But, I see the confusion in setting the headers in the copy-ctor. I will remove the line.